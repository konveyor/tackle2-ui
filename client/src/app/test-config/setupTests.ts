import "@testing-library/jest-dom";
import { server } from "@mocks/server";

const mockInitialized = false;

jest.mock("@react-keycloak/web", () => {
  const originalModule = jest.requireActual("@react-keycloak/web");
  return {
    ...originalModule,
    useKeycloak: () => [mockInitialized],
  };
});

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path",
  }),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// Uncomment the following to see request logging in the console:

// server.events.on("request:start", (req) => {
//   console.log(`Handling a request to ${req.url.href}`);
// });

// server.events.on("request:match", (req) => {
//   console.log(`Request to ${req.url.href} was matched with a handler`);
// });

// server.events.on("request:unhandled", (req) => {
//   console.warn(`Request to ${req.url.href} was not handled`);
// });
