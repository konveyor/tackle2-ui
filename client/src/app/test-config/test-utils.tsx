import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
