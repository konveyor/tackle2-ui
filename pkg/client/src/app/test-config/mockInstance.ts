import MockAdapter from "axios-mock-adapter";
import axios from "axios";

export default new MockAdapter(axios);
