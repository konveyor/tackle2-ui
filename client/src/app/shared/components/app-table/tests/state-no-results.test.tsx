import { render } from "@app/test-config/test-utils";
import React from "react";
import { StateNoResults } from "../state-no-results";

describe("StateNoResults", () => {
  it("Renders without crashing", () => {
    const wrapper = render(<StateNoResults />);
    expect(wrapper).toMatchSnapshot();
  });
});
