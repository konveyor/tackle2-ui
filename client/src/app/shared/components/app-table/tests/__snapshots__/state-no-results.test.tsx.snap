// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateNoResults Renders without crashing 1`] = `
<EmptyState
  variant="small"
>
  <EmptyStateIcon
    icon="test-file-stub"
  />
  <Title
    headingLevel="h2"
    size="lg"
  >
    message.noResultsFoundTitle
  </Title>
  <EmptyStateBody>
    message.noResultsFoundBody
  </EmptyStateBody>
</EmptyState>
`;
